### DISCUSSION TITLE
I accidentally force push my local file

### DISCUSSION THREAD

**Discussion by paraduxos** [community/community#23816] [New to GitHub]
<p>I edit <code>README.md</code> of my repos online (Which was never be pulled to local).</p>
<p>And in my local there were <code>README.md</code> that contain only 1 line.</p>
<p>Then I accidentally <code>force push</code> from local and my online <code>README.md</code> was overwrite from the local one.</p>
<p>Are there anyway to recover this?</p>
<p>I have tried this but I can’t find the HEAD{x} that contains my <code>README.md</code> online</p>
<p><a href="https://github.community/t/i-accidentally-force-pushed-to-my-repo/277">https://github.community/t5/How-to-use-Git-and-GitHub/I-accidentally-force-pushed-to-my-repo/m-p/80#M13</a></p>

**Comment by logankilpatrick** [comment, marked as answer]
<p>Hey <a class="mention" href="/paraduxos">@paraduxos</a>, thanks for being here. Please see here: <a href="https://github.blog/2015-06-08-how-to-undo-almost-anything-with-git/" rel="nofollow noopener">https://github.blog/2015-06-08-how-to-undo-almost-anything-with-git/</a> for a guide on how to undo almost anything with git. </p>
<p>Let me know if you have any follow up questions. </p>

**Comment by tjcreedy** [comment]
<p>Hi, I have exactly the same issue but I can’t seem to figure out what section of this guide should help me.</p>
<p>I made some changes to my README.md on the 18th. I didn’t pull them down to my local repository. I then made some local changes, committed them, then force pushed to the repository (because I’m an idiot, and couldn’t understand why there was a conflict). Now if I look in the history of my README.md online, it shows only the older commits. There is no record of the changes I made. I know that the forced commit was one of two commits, made on the 18th or the 20th.</p>
<p>I tried reverting to one of these commits on my local, but that hasn’t made any difference. I tried cloning the repository and running fsck --full --lost-found, but nothing was found. I can’t see how any of the other suggestions in your link help in this case.</p>
<p>Anything I can do, or do I need to re-do my README.md?</p>

**Comment by airtower-luna** [comment]
<p>I’m afraid you’ll probably have to re-do the README. Overriding remote-only commits on the pushed branch is exactly what a force push does.</p>
<p>I see two other options that might theoretically work, but I have no idea what the chances of them working are:</p>
<ol>
<li>If you know someone who might have fetched the repository in between, the commit might still be around on their system. If anyone complained what the force push was about they’d be a prime candidate. :wink:
</li>
<li>You could try asking GitHub support if they might be able to pull it from a backup.</li>
</ol>
